listener.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from datetime import datetime, timedelta
+
+from apscheduler.events import EVENT_JOB_EXECUTED, JobExecutionEvent
+from apscheduler.schedulers.base import BaseScheduler
+
+
+class JobMisfireError(RuntimeError):
+    pass
+
+
+def add_listener(
```

Comment:
I'd suggest renaming the whole module and function to indicate the purpose: e.g. retry.py